iothub/device/src/ClientFactory.cs

```diff
@@ -499,20 +509,30 @@ internal static ITransportSettings[] GetTransportSettings(TransportType transpor
         /// Ensures that the ClientOptions is configured and initialized.
         /// If a certificate is provided, the fileUploadTransportSettings will use it during initialization.
         /// </summary>
-        private static void EnsureOptionsIsSetup(X509Certificate2 cert, ref ClientOptions options)
+        private static void EnsureOptionsIsSetup(X509Certificate2 cert, IWebProxy proxy, ref ClientOptions options)
         {
-            if (options?.FileUploadTransportSettings == null)
-            {
-                var fileUploadTransportSettings = new Http1TransportSettings { ClientCertificate = cert };
                 if (options == null)
```

Comment:
This is the main fix. If no options, create one. If options has no FileUploadTransportSettings, add one. If cert is provided and ClientCertificate is not null, assign it. Same for proxy.